components/src/hardware-sim/RobotCoordinateSpace/RobotCoordinateSpaceWithDOMCoords.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react'
+import { Svg } from '../../primitives'
+import type { DeckDefinition, DeckSlot } from '@opentrons/shared-data'
+
+export interface RobotCoordinateSpaceWithDOMCoordsRenderProps {
+  deckSlotsById: { [slotId: string]: DeckSlot }
+  getRobotCoordsFromDOMCoords: (
+    domX: number,
+    domY: number
+  ) => { x: number; y: number }
+}
+
+interface RobotCoordinateSpaceWithDOMCoordsProps
+  extends React.ComponentProps<typeof Svg> {
+  viewBox?: string | null
+  deckDef?: DeckDefinition
+  children?: (
+    props: RobotCoordinateSpaceWithDOMCoordsRenderProps
+  ) => React.ReactNode
+}
+
+type GetRobotCoordsFromDOMCoords = RobotCoordinateSpaceWithDOMCoordsRenderProps['getRobotCoordsFromDOMCoords']
+
+export function RobotCoordinateSpaceWithDOMCoords(
+  props: RobotCoordinateSpaceWithDOMCoordsProps
+): JSX.Element | null {
+  const { children, deckDef, viewBox, ...restProps } = props
+  const wrapperRef = React.useRef<SVGSVGElement>(null)
+  const getRobotCoordsFromDOMCoords: GetRobotCoordsFromDOMCoords = (x, y) => {
+    if (!wrapperRef.current) return { x: 0, y: 0 }
+
+    const cursorPoint = wrapperRef.current.createSVGPoint()
+
+    cursorPoint.x = x
+    cursorPoint.y = y
+
+    return cursorPoint.matrixTransform(
+      wrapperRef.current.getScreenCTM()?.inverse()
+    )
+  }
+  if (!deckDef && !viewBox) return null
```

Comment:
deckDef would be fine, but for viewBox, doing null check would be better because of its type.  

```suggestion
  if (deckDef == null && viewBox == null) return null
```